app/src/test/java/io/gnosis/safe/ui/safe/overview/SafeOverviewViewModelTest.kt

```diff
@@ -0,0 +1,62 @@
+package io.gnosis.safe.ui.safe.overview
+
+import android.app.Application
+import android.content.Context
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.spyk
+import io.mockk.verify
+import org.junit.Before
+import org.junit.Test
+import pm.gnosis.svalinn.common.PreferencesManager
+import pm.gnosis.tests.utils.TestPreferences
+
+class SafeOverviewViewModelTest {
+
+    private lateinit var preferences: TestPreferences
+    private lateinit var preferencesManager: PreferencesManager
+    private val context = mockk<Context>(relaxed = true)
+
+    @Before
+    fun setup() {
+        preferences = spyk()
+        val application = mockk<Application>().apply {
+            every { getSharedPreferences(any(), any()) } returns preferences
+        }
+        preferencesManager = PreferencesManager(application)
+    }
+
+    @Test
+    fun `If terms already agreed, advance() method should be called and bottom sheet should not be shown`() {
+        preferences.edit().putBoolean(SafeOverviewViewModel.TERMS_AGREED, true).apply()
+
+        val viewModel = SafeOverviewViewModel(preferencesManager)
+        viewModel.termsBottomSheetDialog = mockk(relaxed = true)
+
+        val callback: () -> Unit = mockk(relaxed = true)
+
+        viewModel.checkTerms(callback)
+
+        verify { callback.invoke() }
+        verify(exactly = 0) { viewModel.termsBottomSheetDialog.show() }
+    }
+
+    @Test
+    fun `If terms not agreed, advance() method should not be called when agree button not hit`() {
+        preferences.edit().putBoolean(SafeOverviewViewModel.TERMS_AGREED, false).apply()
+
+        val viewModel = SafeOverviewViewModel(preferencesManager)
+        viewModel.termsBottomSheetDialog = mockk(relaxed = true)
+        val callback: () -> Unit = mockk(relaxed = true)
+
+        viewModel.checkTerms(callback)
+
+        verify { viewModel.termsBottomSheetDialog.show() }
+        verify(exactly = 0) { callback.invoke() }
+    }
+
+    // TODO: Test if advance is called when user clicks agree
```

Comment:
Should we create an issue for these, or will you add them later to this PR?

Reply:
I was hoping for tips on how to test these in this PR and then I would add them immediately.